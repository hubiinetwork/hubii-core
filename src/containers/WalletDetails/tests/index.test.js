import React from 'react';
import { shallow } from 'enzyme';
import { WalletDetails, mapDispatchToProps } from '../index';

describe('WalletDetails', () => {
  describe('shallow mount', () => {
    const params = {

      match: { params: { address: 'abcd' } },
      history: { location: { pathname: '/wallet/abcd' } },
      currentWalletDetails: {
        balances: [
          { balance: '100', decimals: 2, price: { USD: '1' } },
          { balance: '100', decimals: 2, price: { USD: '1' } },
        ],
      },
    };
    let setCurrentWalletSpy;
    let dom;
    beforeEach(() => {
      setCurrentWalletSpy = jest.fn();
      params.setCurrentWallet = setCurrentWalletSpy;
    });
    describe('#componentDidMount', () => {
      it('should loadWallets action', () => {
        dom = shallow(
          <WalletDetails
            {...params}
          />
        );
        const instance = dom.instance();
        instance.componentDidMount();
        expect(setCurrentWalletSpy).toBeCalledWith(null, params.match.params.address);
      });
    });
    describe('#onHomeClick', () => {
      it('should push history path', () => {
        const historySpy = jest.fn();
        const history = { push: historySpy, location: { pathname: '/wallet/abcd' } };
        dom = shallow(
          <WalletDetails
            {...params}
            history={history}
          />
        );
        const instance = dom.instance();
        instance.onHomeClick();
        expect(historySpy).toBeCalledWith('/wallets');
      });
    });
    describe('#onTabsChange', () => {
      it('should push history path', () => {
        const historySpy = jest.fn();
        const history = { push: historySpy, location: { pathname: '/wallet/abcd' } };
        dom = shallow(
          <WalletDetails
            {...params}
            history={history}
          />
        );
        const instance = dom.instance();
        const path = '/path';
        instance.onTabsChange(path);
        expect(historySpy).toBeCalledWith(path);
      });
    });
    describe('#render', () => {
      it('should show total usd value in header', () => {
        dom = shallow(
          <WalletDetails
            {...params}
          />
        );
        expect(dom.find('WalletHeader').props().balance).toEqual(2);
      });
    });
    describe('#mapDispatchToProps', () => {
      it('should dispatch', () => {
        const dispatchSpy = jest.fn();
        const actions = mapDispatchToProps(dispatchSpy);
        Object.keys(actions).forEach((action, index) => {
          actions[action]();
          expect(dispatchSpy).toHaveBeenCalledTimes(index + 1);
        });
      });
    });
  });
});
