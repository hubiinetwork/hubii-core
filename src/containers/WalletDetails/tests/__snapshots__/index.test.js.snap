// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletDetails should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletDetails
    currentWalletDetails={
        Immutable.Map {
            name: "1",
            address: "0x910c4BA923B2243dc13e00A066eEfb8ffd905EB0",
            type: "software",
            encrypted: "{\\"address\\":\\"910c4ba923b2243dc13e00a066eefb8ffd905eb0\\",\\"id\\":\\"ae1eb0e5-f3e6-4bc8-b23c-1b6efb8e7f4e\\",\\"version\\":3,\\"Crypto\\":{\\"cipher\\":\\"aes-128-ctr\\",\\"cipherparams\\":{\\"iv\\":\\"160194dccdc2e29f98bc0b02f811dc95\\"},\\"ciphertext\\":\\"1e787b400e5fc733590fbf3a4cd547e81a1004cd3c7403f68416be472500915a\\",\\"kdf\\":\\"scrypt\\",\\"kdfparams\\":{\\"salt\\":\\"8adc2aed2709e8de54ba69903d102f25c8d87920f1f5cd952fa0c000aa7c91ac\\",\\"n\\":131072,\\"dklen\\":32,\\"p\\":1,\\"r\\":8},\\"mac\\":\\"952bcd9627ada0a96e83dd9feb4bdc6c47a39f4d20db0e9d649f6b247e5538ff\\"}}",
            transactions: Immutable.Map {
                loading: true,
                transactions: Immutable.List [
                ],
            },
            balances: Immutable.Map {
                loading: false,
                error: null,
                assets: Immutable.List [
                    Immutable.Map {
                        balance: "0.29972769885",
                        currency: "ETH",
                        symbol: "ETH",
                        value: Immutable.Map {
                            eth: "0.29972769885",
                            btc: "0.0029972769885",
                            usd: "123.4878119262",
                        },
                    },
                    Immutable.Map {
                        balance: "1",
                        currency: "0x583cbbb8a8443b38abcc0c956bece47340ea1367",
                        symbol: "BOKKY",
                        value: Immutable.Map {
                            eth: "0",
                            btc: "0",
                            usd: "0",
                        },
                    },
                ],
                total: Immutable.Map {
                    eth: "0.29972769885",
                    btc: "0.0029972769885",
                    usd: "123.4878119262",
                },
            },
            loadingBalances: false,
            loadingBalancesError: null,
        }
    }
    history={
        Object {
            "location": Object {
              "pathname": "/wallet/abcd",
            },
          }
    }
    match={
        Object {
            "params": Object {
              "address": "abcd",
            },
          }
    }
    setCurrentWallet={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <index.style__TabsLayout>
          <WalletHeader
                    address="0x910c4BA923B2243dc13e00A066eEfb8ffd905EB0"
                    balance={123.4878119262}
                    iconType="home"
                    name="1"
                    onIconClick={[Function]}
          />
</index.style__TabsLayout>,
        <Tab.style
          activeKey="/wallet/abcd"
          animated={false}
          onChange={[Function]}
>
          <TabPane
                    placeholder={null}
                    tab={
                              <span>
                                        <Icon
                                                  type="wallet"
                                        />
                                        Overview
                              </span>
                    }
          >
                    <Route
                              component={[Function]}
                              path="undefined/overview"
                    />
          </TabPane>
          <TabPane
                    placeholder={null}
                    tab={
                              <span>
                                        <Icon
                                                  type="contacts"
                                        />
                                        Transfer
                              </span>
                    }
          >
                    <Route
                              component={[Function]}
                              path="undefined/transfer"
                    />
          </TabPane>
          <TabPane
                    placeholder={null}
                    tab={
                              <span>
                                        <Icon
                                                  type="shopping-cart"
                                        />
                                        Buy ETH
                              </span>
                    }
          >
                    <Route
                              component={[Function]}
                              path="undefined/buyeth"
                    />
          </TabPane>
</Tab.style>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WalletHeader
            address="0x910c4BA923B2243dc13e00A066eEfb8ffd905EB0"
            balance={123.4878119262}
            iconType="home"
            name="1"
            onIconClick={[Function]}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "address": "0x910c4BA923B2243dc13e00A066eEfb8ffd905EB0",
            "balance": 123.4878119262,
            "iconType": "home",
            "name": "1",
            "onIconClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeKey": "/wallet/abcd",
          "animated": false,
          "children": Array [
            <TabPane
              placeholder={null}
              tab={
                            <span>
                                          <Icon
                                                        type="wallet"
                                          />
                                          Overview
                            </span>
              }
>
              <Route
                            component={[Function]}
                            path="undefined/overview"
              />
</TabPane>,
            <TabPane
              placeholder={null}
              tab={
                            <span>
                                          <Icon
                                                        type="contacts"
                                          />
                                          Transfer
                            </span>
              }
>
              <Route
                            component={[Function]}
                            path="undefined/transfer"
              />
</TabPane>,
            <TabPane
              placeholder={null}
              tab={
                            <span>
                                          <Icon
                                                        type="shopping-cart"
                                          />
                                          Buy ETH
                            </span>
              }
>
              <Route
                            component={[Function]}
                            path="undefined/buyeth"
              />
</TabPane>,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "undefined/overview",
            "nodeType": "class",
            "props": Object {
              "children": <Route
                component={[Function]}
                path="undefined/overview"
/>,
              "placeholder": null,
              "tab": <span>
                <Icon
                                type="wallet"
                />
                Overview
</span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "undefined/overview",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "undefined/transfer",
            "nodeType": "class",
            "props": Object {
              "children": <Route
                component={[Function]}
                path="undefined/transfer"
/>,
              "placeholder": null,
              "tab": <span>
                <Icon
                                type="contacts"
                />
                Transfer
</span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "undefined/transfer",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "undefined/buyeth",
            "nodeType": "class",
            "props": Object {
              "children": <Route
                component={[Function]}
                path="undefined/buyeth"
/>,
              "placeholder": null,
              "tab": <span>
                <Icon
                                type="shopping-cart"
                />
                Buy ETH
</span>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "undefined/buyeth",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <index.style__TabsLayout>
            <WalletHeader
                        address="0x910c4BA923B2243dc13e00A066eEfb8ffd905EB0"
                        balance={123.4878119262}
                        iconType="home"
                        name="1"
                        onIconClick={[Function]}
            />
</index.style__TabsLayout>,
          <Tab.style
            activeKey="/wallet/abcd"
            animated={false}
            onChange={[Function]}
>
            <TabPane
                        placeholder={null}
                        tab={
                                    <span>
                                                <Icon
                                                            type="wallet"
                                                />
                                                Overview
                                    </span>
                        }
            >
                        <Route
                                    component={[Function]}
                                    path="undefined/overview"
                        />
            </TabPane>
            <TabPane
                        placeholder={null}
                        tab={
                                    <span>
                                                <Icon
                                                            type="contacts"
                                                />
                                                Transfer
                                    </span>
                        }
            >
                        <Route
                                    component={[Function]}
                                    path="undefined/transfer"
                        />
            </TabPane>
            <TabPane
                        placeholder={null}
                        tab={
                                    <span>
                                                <Icon
                                                            type="shopping-cart"
                                                />
                                                Buy ETH
                                    </span>
                        }
            >
                        <Route
                                    component={[Function]}
                                    path="undefined/buyeth"
                        />
            </TabPane>
</Tab.style>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WalletHeader
              address="0x910c4BA923B2243dc13e00A066eEfb8ffd905EB0"
              balance={123.4878119262}
              iconType="home"
              name="1"
              onIconClick={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "address": "0x910c4BA923B2243dc13e00A066eEfb8ffd905EB0",
              "balance": 123.4878119262,
              "iconType": "home",
              "name": "1",
              "onIconClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeKey": "/wallet/abcd",
            "animated": false,
            "children": Array [
              <TabPane
                placeholder={null}
                tab={
                                <span>
                                                <Icon
                                                                type="wallet"
                                                />
                                                Overview
                                </span>
                }
>
                <Route
                                component={[Function]}
                                path="undefined/overview"
                />
</TabPane>,
              <TabPane
                placeholder={null}
                tab={
                                <span>
                                                <Icon
                                                                type="contacts"
                                                />
                                                Transfer
                                </span>
                }
>
                <Route
                                component={[Function]}
                                path="undefined/transfer"
                />
</TabPane>,
              <TabPane
                placeholder={null}
                tab={
                                <span>
                                                <Icon
                                                                type="shopping-cart"
                                                />
                                                Buy ETH
                                </span>
                }
>
                <Route
                                component={[Function]}
                                path="undefined/buyeth"
                />
</TabPane>,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "undefined/overview",
              "nodeType": "class",
              "props": Object {
                "children": <Route
                  component={[Function]}
                  path="undefined/overview"
/>,
                "placeholder": null,
                "tab": <span>
                  <Icon
                                    type="wallet"
                  />
                  Overview
</span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "undefined/overview",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "undefined/transfer",
              "nodeType": "class",
              "props": Object {
                "children": <Route
                  component={[Function]}
                  path="undefined/transfer"
/>,
                "placeholder": null,
                "tab": <span>
                  <Icon
                                    type="contacts"
                  />
                  Transfer
</span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "undefined/transfer",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "undefined/buyeth",
              "nodeType": "class",
              "props": Object {
                "children": <Route
                  component={[Function]}
                  path="undefined/buyeth"
/>,
                "placeholder": null,
                "tab": <span>
                  <Icon
                                    type="shopping-cart"
                  />
                  Buy ETH
</span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "undefined/buyeth",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
